### PR TITLE
PFR v3

### PR DESCRIPTION
**Pull request by bleemb**
#58 
**Auditors & Unit tests added**
ec2: AMI age < 6 months, AMI correctly registered
Shield: Any attacks in last week
ACM: Certs Renewed correctly, certs valid
CloudHSM: Clusters in degraded state, HSMs in degraded state, At least 1 cluster backup in READY state
IAM: Least privilege for Customer managed Policies, Users, Roles, Groups

### CODE REVIEW

File: eeauditor/auditors/aws/Amazon_EC2_Auditor.py [hunk 1]
```diff
@@ -788,4 +789,421 @@ def ec2_serial_console_access_check(cache: dict, awsAccountId: str, awsRegion: s
...
             "RecordState": "ARCHIVED"
         }
-        yield finding
\ No newline at end of file
+        yield finding
+
+
+@registry.register_check("ec2")
+def ec2_ami_age_check(cache: dict, awsAccountId: str, awsRegion: str, awsPartition: str) -> dict:
+    """[EC2.5] EC2 Instances should use AMIs that are less than 6 months old"""
```
Comment (jonrau1) [inline eeauditor/auditors/aws/Amazon_EC2_Auditor.py:797]: Can we switch this one to 3 months? Figure most companies doing this seriously are baking golden images every 2 or 4 weeks, and may have a N - 2 version policy

**Review by jonrau1 — COMMENTED** [review]
Just minor changes on some of the specs, overall really solid!

File: eeauditor/auditors/aws/Amazon_EC2_Auditor.py (hunk 1 above)
Comment (bleemb) [reply eeauditor/auditors/aws/Amazon_EC2_Auditor.py]: Absolutely, I'm all for putting it lower!  Lowered it to 3 months :)

**Comment by bleemb** [comment]
Thanks for that feedback @jonrau1 - I've gone through and updated the ASFF schema to use the relevant resource types :)

**Pull request merged**